src/storage/heap_file.c

```diff
@@ -23233,6 +23236,24 @@ heap_insert_hfid_for_class_oid (THREAD_ENTRY * thread_p, const OID * class_oid,
   assert (entry->hfid.hpgid == NULL_PAGEID);
 
   HFID_COPY (&entry->hfid, hfid);
```

Comment:
Please add last argument to lf_hash_find_or_insert that outputs true/false if entry is inserted or not (if an entry already existed, a new one is not inserted). Then add a safe-guard that entry is inserted for the first time. Otherwise, classname may be overwritten and memory can be leaked.

Reply:
heap_cache_class_info does not suffer of a race to set class name. It is called when heap file is created or on boot. no concurrent threads that tries to read/write same entry. Therefore, class name can be set directly.

A comment may help explaining.